lib/IncomingStream.js

```
@@ -279,6 +279,32 @@ class IncomingStream extends Emitter
 		!this.stopped && this.emit("track",this,incomingStreamTrack);
 	}
 	
+	/**
+	 * Remove a track from stream. Note the removed track is not stopped by calling this
+	 * function. It's the caller's responsibility to stop it if the track is not used by
+	 * any stream any more.
+	 * 
+	 * @param {string} trackId 
+	 */
+	removeTrack(trackId)
+	{
+		let incomingStreamTrack = this.tracks.get(trackId);
+		if (incomingStreamTrack)
```

should we throw or return false if trackId is not found?

---

You have changed to return the removed track. I think that is good now as it may return undefined when the trackId is not found. One reference is probably the JS Map class, which wouldn't throw if there is no key.